### ISSUE TITLE
Nightly test job "c2chapel" broke

### ISSUE THREAD

**Issue by awallace-cray** [chapel-lang/chapel#8677]
2018-03-01 : Jenkins nightly c2chapel test started failing on 02/23 (`correctness-test-c2chapel/213`), after some unknown change on 02/22 broke all https connections to GitHub URLs from several common client programs (git, wget, curl) running on Cray internal (I.T.-built) SLES-11 systems.  
```
+ cd tools/c2chapel
+ make
Using system virtualenv
mkdir -p /ptmp/jenkins/chapel-ci/workspace/correctness-test-c2chapel/tools/c2chapel/install/venv
mkdir -p /ptmp/jenkins/chapel-ci/workspace/correctness-test-c2chapel/tools/c2chapel/install/fakeHeaders
cd /ptmp/jenkins/chapel-ci/workspace/correctness-test-c2chapel/tools/c2chapel/install && \
	echo "Fetching release 2.17 tarball..." && \
	wget https://github.com/eliben/pycparser/archive/release_v2.17.tar.gz -O release_v2.17.tar.gz && \
	tar xzf release_v2.17.tar.gz -C /ptmp/jenkins/chapel-ci/workspace/correctness-test-c2chapel/tools/c2chapel/install/fakeHeaders --strip 3 pycparser-release_v2.17/utils/fake_libc_include
Fetching release 2.17 tarball...
--2018-02-23 02:00:02--  https://github.com/eliben/pycparser/archive/release_v2.17.tar.gz
Resolving github.com... 192.30.255.112, 192.30.255.113
Connecting to github.com|192.30.255.112|:443... connected.
OpenSSL: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version
Unable to establish SSL connection.
make: *** [c2chapel] Error 1
```

Some other nightly test jobs broke for similar reasons, but they were successfully moved to run on other Cray systems which did not have this GitHub/https connection problem.  

However, I was never able to get this c2chapel test job to run on any non SLES-11 system. Systems tried:
* chapcs10 (SLES-12, Cray-I.T.-built): with or without system-installed virtualenv and pip, see PR #8668
  Chapel build (make and make check) ran ok, after minor kludging; however, the SKIPIF fired, so no tests ran, which is an error
```
[Working on directory /ptmp/chapelu/jenkins/chapel-ci.bogus/chapcs10x/workspace/correctness-test-c2chapel/test/c2chapel]
[Skipping directory based on SKIPIF environment settings]
[Removing /tmp/chplTestTmpDir.pfF5P8 directory]
[Done with tests - 180226.103037]
[Log file: /tmp/day1-Mon-c2chapel.log.30463 ]

2018-02-26 10:30:37,422 [INFO] Parsed 0 test cases from "/tmp/day1-Mon-c2chapel.log.30463".
2018-02-26 10:30:37,422 [INFO] Wrote jUnit report to: /ptmp/chapelu/jenkins/chapel-ci.bogus/chapcs10x/workspace/correctness-test-c2chapel/chapel-tests.xml
[Generating jUnit XML report]
[Test Summary - 180226.103037]
[Summary: #Successes = 0 | #Failures = 0 | #Futures = 0 | #Warnings = 0 ]
[Summary: #Passing Suppressions = 0 | #Passing Futures = 0 ]
[END]
...
2018-02-26 10:30:37 [INFO] Finished running ./util/cron/test-c2chapel.bash on chapcs10
Recording test results
ERROR: Step ‘Publish JUnit test result report’ failed: None of the test reports contained any result
```
* chapel-ubu1604-01 (Ubuntu-64 - plain-vanilla VM, not Cray I.T.-built)
  Ubuntu 16.04 wants to use a more primitive version of "sh" than other systems.  I first saw this immediately in Jenkins itself- the "source" command used in this Jenkins job config was "not found" ("execute shell" build step). 
```
[correctness-test-c2chapel] $ /bin/sh -xe /tmp/jenkins1428314992472207791.sh
+ set -x
+ source ./util/setchplenv.sh
/tmp/jenkins1428314992472207791.sh: 3: /tmp/jenkins1428314992472207791.sh: source: not found
Build step 'Execute shell' marked build as failure
```
Hacked through the same basic problem in other shell snippets, eventually including the SHELL used by make itself, by changing all "/bin/sh" (expressed or implied) to "/bin/bash".  
Finally, stopped trying.
```
. /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/tools/c2chapel/install/venv/bin/activate && \
  pip install /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/tools/c2chapel/install/release_v2.17.tar.gz && \
  pip install argparse
Processing ./install/release_v2.17.tar.gz
Installing collected packages: pycparser
  Running setup.py install for pycparser: started
    Running setup.py install for pycparser: finished with status 'done'
Successfully installed pycparser-2.17
Requirement already satisfied: argparse in /usr/lib/python2.7
+ make SHELL=/bin/bash install
Installing c2chapel...
Installing symbolic link...
mkdir -p /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/bin/linux64
ln -s /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/tools/c2chapel/c2chapel /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/bin/linux64/c2chapel
+ make SHELL=/bin/bash check
cd test && ./tester.sh
Testing c2chapel...\n
No arguments: ERROR
"c2chapel " vs. no-args.good':
1,2c1,4
< /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/bin/linux64/c2chapel: 24: /jenkins/chapel-ci.bogus/workspace/correctness-test-c2chapel/bin/linux64/c2chapel: source: not found
< Unable to import pycparser: No module named pycparser
---
> usage: c2chapel [-h] [--no-typedefs] [--debug] [--no-fake-headers]
>                 [--no-comments] [-V]
>                 file [cppFlags [cppFlags ...]]
> c2chapel: error: too few arguments
--help: ERROR
```

To sum up, this test config:
* Cannot run on SLES-12, at least not yet (@benharsh any update)? 
* I could not get it to run on Ubuntu-16.04 because so many shell snippets (incl Makefile rules) used bash-like constructs that Ubuntu's default "sh" would not accept. 
* (It also cannot run on SLES-11, at least not Cray's internal SLES-11 systems- but blame that on Cray and/or GitHub, not this test config)

Likely next step: see if this test config will run on chapelmac, chapel-centos7-01, etc.

**Comment by awallace-cray** [comment]
2018-03-01 : Record the change-`sh`-to-`bash` hacks I tried on Ubuntu.

* Jenkins execute shell job step: 
  * add `#!/bin/bash` to the top
  * add `SHELL=/bin/bash` to "make" command lines.
* chapel.git
```
diff --git a/tools/c2chapel/test/tester.sh b/tools/c2chapel/test/tester.sh
index e4c1ab2..ab3be9e 100755
--- a/tools/c2chapel/test/tester.sh
+++ b/tools/c2chapel/test/tester.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
 
 # This script must be run within c2chapel/test
 
@@ -70,4 +70,5 @@ if [ "$numFailures" -eq "0" ]; then
   printf "${GREEN}SUCCESS${NORMAL}\n"
 else
   printf "${RED}$numFailures/$numTotal tests failed${NORMAL}\n"
+  exit 1
 fi
```
  Repro the Ubuntu make/sh issue:
```
# chapelu@chapel-ubu1604-01:~$ cat Makefile
PHONY: foo bar
foo:
	echo SHELL=$(SHELL)=$$SHELL
bar:
	source noway

# chapelu@chapel-ubu1604-01:~$ make bar
source noway
make: source: Command not found
Makefile:5: recipe for target 'bar' failed
make: *** [bar] Error 127

# chapelu@chapel-ubu1604-01:~$ make SHELL=/bin/bash bar
source noway
/bin/bash: noway: No such file or directory
Makefile:5: recipe for target 'bar' failed
make: *** [bar] Error 1
```

**Comment by awallace-cray** [comment]
Note: Forgot to mention, `tools/c2chapel/test/tester.sh` needs an `exit 1` or similar, otherwise it will never return error status to whoever calls it. See above git diff.

**Comment by npadmana** [comment]
@awallace-cray -- #8662 may partially address this.

**Comment by awallace-cray** [comment]
Ben's PR #8738 fixes this, except:
- not the original SLES-11 https GH connection problem, so we can't run it on SLES-11 any more
- not the Ubuntu-related shell problems noted above, so we can't run it on Ubuntu
We can run this job on the chapel-centos7-01 VM.
